specification/operationalinsights/resource-manager/Microsoft.OperationalInsights/preview/2020-03-01-preview/AvailableServiceTiers.json

```diff
@@ -0,0 +1,138 @@
+{
+  "swagger": "2.0",
+  "info": {
+    "title": "Azure Log Analytics",
+    "description": "Azure Log Analytics API reference",
+    "version": "2020-03-01-preview"
+  },
+  "host": "management.azure.com",
+  "schemes": [
+    "https"
+  ],
+  "consumes": [
+    "application/json"
+  ],
+  "produces": [
+    "application/json"
+  ],
+  "security": [
+    {
+      "azure_auth": [
+        "user_impersonation"
+      ]
+    }
+  ],
+  "securityDefinitions": {
+    "azure_auth": {
+      "type": "oauth2",
+      "authorizationUrl": "https://login.microsoftonline.com/common/oauth2/authorize",
+      "flow": "implicit",
+      "description": "Azure Active Directory OAuth2 Flow",
+      "scopes": {
+        "user_impersonation": "impersonate your user account"
+      }
+    }
+  },
+  "paths": {
+    "/subscriptions/{subscriptionId}/resourcegroups/{resourceGroupName}/providers/Microsoft.OperationalInsights/workspaces/{workspaceName}/availableServiceTiers": {
+      "get": {
+        "tags": [
+          "AvailableServiceTiers"
+        ],
+        "x-ms-examples": {
+          "AvailableServiceTiers": {
+            "$ref": "./examples/WorkspacesAvailableServiceTiers.json"
+          }
+        },
+        "operationId": "Workspaces_AvailableServiceTiers",
+        "description": "Gets the available service tiers for the workspace.",
+        "parameters": [
+          {
+            "$ref": "../../../../../common-types/resource-management/v1/types.json#/parameters/SubscriptionIdParameter"
+          },
+          {
+            "$ref": "../../../../../common-types/resource-management/v1/types.json#/parameters/ResourceGroupNameParameter"
+          },
+          {
+            "$ref": "../../../common/v1/types.json#/parameters/WorkspaceNameParameter"
+          },
+          {
+            "$ref": "../../../../../common-types/resource-management/v1/types.json#/parameters/ApiVersionParameter"
+          }
+        ],
+        "responses": {
+          "200": {
+            "description": "OK response definition.",
+            "schema": {
+              "type": "array",
+              "items": {
+                "$ref": "#/definitions/AvailableServiceTier"
+              }
+            }
+          }
+        }
+      }
+    }
+  },
+  "definitions": {
+    "AvailableServiceTier": {
```

Comment:
The response body should follow the standard envelope/schema for proxyOnly resources:
```id, type, name, properties{your propertes}```

Additionally, the response should look like:
```
    "properties": {
    "value": {
        "type": "array",
        "items": {
            <<your item >>
        },
        "description": "An array of resources."
    },
```

Reply:
this api was cloned from our stable version. we can't change that

Reply:
This API is already in use. I added a work item to followup and fix.

Reply:
@yoramsinger @arolshan  - I sincerely appreciate your commitment to address the modeling errors in the APIs in a future update. For the present concern, what is being changed in this API? Is it an exact fork of an existing API into a new version? I'm trying to understand what to review. Feel free to message or email me if that's easier.

Btw, it creates a cleaner diff if the first commit of a new API version is the exact contents of a previous version but with new file names. Subsequent commits would bring the changes. This enables an inline diff in the PR (by selecting the later commits) that shows just changes that are the target for review.

Reply:
Hi @KrisBash, every **new** json file in this PR is a clone of an api path in the OperationalInsights.json file from the 2015 stable version. 
We separated them to separate files, so it would be easier to incrementally fix them and apply changes (in new api versions) in the future without breaking the sdk generation. 
in addition, I did work to use only common-types where possible.